### PR TITLE
feat: hide invite option based on squad permission

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- Hide invitation button from squad header
- Hide invitation CTAa from members modal
- Hide invitation CTA from modals (unused but just to be sure)
- API implementation https://github.com/dailydotdev/daily-api/pull/1240

Also implements WT-1267

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1266 #done

### CODE REVIEW

File: packages/shared/src/components/modals/SquadInviteModal.tsx [hunk 1]
```diff
@@ -50,15 +50,17 @@ function SquadInviteModal({
...
-      <Modal.Footer>
-        <Button
-          icon={<LinkIcon />}
-          className="flex-1 mx-4 btn-primary-cabbage"
-          onClick={() => inviteTextRef.current?.copyLink()}
-        >
-          Copy invitation link
-        </Button>
-      </Modal.Footer>
+      {verifyPermission(squad, SourcePermissions.Invite) && (
```
Comment (rebelchris) [inline packages/shared/src/components/modals/SquadInviteModal.tsx:53]: Wait, but we shouldn't even show this modal if you don't have the permission right?
Not sure this step is needed here.

File: packages/shared/src/components/modals/SquadMemberModal.tsx [hunk 2]
```diff
@@ -36,6 +36,10 @@ const InitialItem = ({ squad }: { squad: Squad }) => {
     origin: Origin.SquadMembersList,
   });
 
+  if (!verifyPermission(squad, SourcePermissions.Invite)) {
+    return null;
+  }
```
Comment (rebelchris) [inline packages/shared/src/components/modals/SquadMemberModal.tsx:41]: should this not go above the hook, just so we don't load it unneeded?
Or is that a anti-pattern?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice, got some minor cleanup things I think we can do.

File: packages/shared/src/components/modals/SquadMemberModal.tsx (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:41]: Yes

File: packages/shared/src/components/modals/SquadInviteModal.tsx (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/components/modals/SquadInviteModal.tsx:53]: @rebelchris It is actually unused so we can remove component altogether?

File: packages/shared/src/components/modals/SquadMemberModal.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:41]: Ah yes, we don't have locked squad anymore, so please do a cleanup yes 🙏

File: packages/shared/src/components/modals/SquadInviteModal.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/modals/SquadInviteModal.tsx:53]: Ah yes, we don't have locked squad anymore, so please do a cleanup yes 🙏

**Review by rebelchris — APPROVED** [review]
Nice work! 🙏

**Pull request merged**